concurrency-primer.tex

```diff
@@ -883,6 +883,94 @@ \subsection{Conclusion about lock-free}
 Balancing complexity and performance is essential in concurrency, 
 a domain fraught with challenges.
 
+\subsection{ABA problem}
+We have introduced CAS as one of the read-modify-write operations.
```

Comment:
Proofread as following:
> CAS has been introduced as one of the read-modify-write operations.
> However, the target object not changing does not necessarily mean that no other threads modified it halfway through.
> If the target object is changed by another thread and then changed back, the result of comparison would still be equal. In this case, the target object has indeed been modified, yet the operation appears unchanged, compromising its atomicity.

The above presents them as statements rather than questions.